api/src/glfs-fops.c

```diff
@@ -516,6 +516,165 @@ pub_glfs_open(struct glfs *fs, const char *path, int flags)
     return glfd;
 }
 
+static inline void
+cleanup_fopat_args(struct glfs_fd *pglfd, xlator_t *subvol, int ret,
+                   dict_t *dict, loc_t *loc)
+{
+    if (dict)
+        dict_unref(dict);
+
+    if (loc)
+        loc_wipe(loc);
+
+    if (subvol)
+        glfs_subvol_done(pglfd->fs, subvol);
+
+    GF_REF_PUT(pglfd);
+}
+
+xlator_t *
+setup_fopat_args(struct glfs_fd *pglfd, const char *path, int flags, loc_t *loc,
+                 dict_t *xattr, struct iatt *iatt)
+{
+    int ret = 0;
+    gf_boolean_t no_follow = false;
+    xlator_t *subvol = NULL;
+
+    GF_REF_GET(pglfd);
+
+    subvol = glfs_active_subvol(pglfd->fs);
+    if (!subvol) {
+        ret = -1;
+        errno = EIO;
+        goto out;
+    }
+
+    no_follow = (flags & AT_SYMLINK_NOFOLLOW) == AT_SYMLINK_NOFOLLOW;
+    if (flags & AT_SYMLINK_FOLLOW)
+        no_follow = false;
+
+    if (xattr) {
```

Comment:
I thought this was used by all fops, but an overview of some other *at changes seems to indicate that it's not used everywhere. For now I think it will be better to keep it as it's now and maybe propose changes on follow up patches.